src/redux-enhancers/undoRedoEnhancer.ts

```diff
@@ -175,12 +175,9 @@ const undoReducer = (state: State, undoPatches: Patch[]) => {
 
   if (!undoPatches.length) return state
 
-  const undoTwice =
-    penultimateUndoPatch &&
-    !!(
-      (lastAction && NAVIGATION_ACTIONS[lastAction]) ||
-      (penultimateAction && (NAVIGATION_ACTIONS[penultimateAction] || penultimateAction === 'newThought'))
-    )
+  const undoTwice = NAVIGATION_ACTIONS[lastAction]
+    ? penultimateUndoPatch && (penultimateAction ? UNDOABLE_ACTIONS[penultimateAction] : false)
+    : penultimateUndoPatch && penultimateAction === 'newThought'
```

Comment:
There are some simplifications we can make to the logic here.

`a ? b : false` is equivalent to `a && b`:

```ts
const undoTwice = NAVIGATION_ACTIONS[lastAction]
    ? penultimateUndoPatch && (penultimateAction && UNDOABLE_ACTIONS[penultimateAction])
    : penultimateUndoPatch && penultimateAction === 'newThought'
```

We can simplify `penultimateAction && UNDOABLE_ACTIONS[penultimateAction]` to `UNDOABLE_ACTIONS[penultimateAction]` since the penultimate action must exist if it is in `UNDOABLE_ACTIONS`:

```ts
const undoTwice = NAVIGATION_ACTIONS[lastAction]
    ? penultimateUndoPatch && UNDOABLE_ACTIONS[penultimateAction]
    : penultimateUndoPatch && penultimateAction === 'newThought'
```

Finally, we can remove `penultimateUndoPatch` from the condition since it must be true whenever there is a `penultimateAction`:

```ts
  const undoTwice = NAVIGATION_ACTIONS[lastAction]
    ? UNDOABLE_ACTIONS[penultimateAction]
    : penultimateAction === 'newThought'
```